xlml/utils/gpu.py

```diff
@@ -304,48 +124,163 @@ def create_resource(
   Returns:
     The ip address of the GPU VM.
   """
-  image = get_image_from_family(project=image_project, family=image_family)
-  disk_type = f"zones/{gcp.zone}/diskTypes/pd-ssd"
-  disks = [disk_from_image(disk_type, 100, True, image.self_link)]
-  metadata = create_metadata({
-      # "install-nvidia-driver": "True",
-      "install-nvidia-driver": "False",
-      "proxy-mode": "project_editors",
-      "ssh-keys": f"cloud-ml-auto-solutions:{ssh_keys.public}",
-  })
-  acceleratorConfig = compute_v1.AcceleratorConfig(
-      accelerator_count=accelerator.count,
-      accelerator_type=f"projects/{gcp.project_name}/zones/{gcp.zone}/acceleratorTypes/{accelerator.accelerator_type}",
-  )
-  service_account = compute_v1.ServiceAccount(
-      # email = "cloud-auto-ml-solutions@google.com",
-      scopes=["https://www.googleapis.com/auth/cloud-platform"]
-  )
-  instance = create_instance(
-      project_id=gcp.project_name,
-      zone=gcp.zone,
-      instance_name=gpu_name,
-      disks=disks,
-      machine_type=accelerator.machine_type,
-      service_account=service_account,
+  project_id = gcp.project_name
+  zone = gcp.zone
+
+  @task
+  def create_resource_request(
+      instance_name: str,
+      accelerator: test_config.Gpu,
+      ssh_keys: ssh.SshKeys,
+      instance_termination_action: str,
       external_access=True,
-      accelerators=[acceleratorConfig],
-      metadata=metadata,
+      spot: bool = False,
+      delete_protection: bool = False,
+  ) -> airflow.XComArg:
+    """
+    Send an instance creation request to the Compute Engine API and wait for it to complete.
+
+    Args:
+        instance_name: name of the new virtual machine (VM) instance.
+        accelerator: Description of GPU to create.
+        ssh_keys: XCom value for SSH keys to communicate with these GPUs.
+        instance_termination_action: What action should be taken once a Spot VM is terminated.
+            Possible values: "STOP", "DELETE"
+        external_access: boolean flag indicating if the instance should have an external IPv4
+            address assigned.
+        spot: boolean value indicating if the new instance should be a Spot VM or not.
+        delete_protection: boolean value indicating if the new virtual machine should be
+            protected against deletion or not.
+    Returns:
+        Ip address of the instance object created.
+    """
+    machine_type = accelerator.machine_type
+    image = get_image_from_family(project=image_project, family=image_family)
+    disk_type = f"zones/{gcp.zone}/diskTypes/pd-ssd"
+    disks = [disk_from_image(disk_type, 100, True, image.self_link)]
+    metadata = create_metadata({
+        # "install-nvidia-driver": "True",
```

Comment:
Do we want to use true or false? If false is enough, could we remove this comment for less confusion?

Reply:
Right, removed!